src/middleware.js

```diff
@@ -30,7 +30,8 @@ export default function middleware(options = {}) {
     Object.assign(req, {span});
 
     // finalize the span when the response is completed
-    res.on("finish", () => {
+    const endCopy = res.end;
+    res.end = function() {
```

Comment:
Please don't monkey-patch `res.end`
Use `res.on('close', cb)` and `res.on('finish', cb)`